src/internal/testutil/kubeclient.go

```diff
@@ -210,3 +244,36 @@ func WaitForPachdReady(t testing.TB, namespace string) {
 		}
 	}
 }
+
+// Wait deployment only works for deployments and checks every pod.
+// This function is simpler and just unblocks as soon as 1 pod is read.
+// It can be used on non-Deployment objects like StatefulSets.
+func WaitForFirstPodReady(t testing.TB, appName string, namespace string) {
+	k := GetKubeClient(t)
+	watch, err := k.CoreV1().Pods(namespace).Watch(context.Background(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", appName),
```

Comment:
This should probably be refactored to take a selector instead of hard-coding app.  The tests for pipelines below really want to supply pipelineName=<genreated name> and not app=pipeline (which applies to all pipelines, even ones that were started by a previous test when -clusters.reuse is true).